### PR TITLE
Null/empty check username on join and chat use

### PR DESCRIPTION
**Pull request by ThatDan123**
Check for null/empty name on join and chat use

[Edit] Just testing now

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs [hunk 1]
```diff
@@ -310,44 +310,40 @@ public bool IsMentor(string userID)
...
-		ConnectedPlayer unverifiedConnPlayer,
-		string unverifiedClientId)
+	private async Task<bool> CheckUserState(string unverifiedToken, ConnectedPlayer unverifiedConnPlayer)
 	{
+		//Allow local offline testing
 		if (GameData.Instance.OfflineMode)
 		{
 			Logger.Log($"{unverifiedConnPlayer.Username} logged in successfully in offline mode. " +
-					   $"userid: {unverifiedUserid}", Category.Admin);
+					   $"userid: {unverifiedConnPlayer.UserId}", Category.Admin);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs:319]: you tested with build?

File: UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs [hunk 2]
```diff
@@ -383,45 +381,54 @@ public bool IsMentor(string userID)
 			if (response == null) return false;
 		}
 
-		//allow error response for local offline testing
+		//Must have non-null/empty username
+		if (string.IsNullOrEmpty(unverifiedConnPlayer.Username))
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs:385]: oh yeah, that reminds me we have no way of validating Username, that needs to be added at some point

**Review by Bod9001 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs:319]: erm, nope will do tomorrow.

**Review by NoooneyDude — APPROVED** [review]

**Comment by corp-0** [comment]
uhm this has two approvals but also a waiting for test label and unresolved conversations

**Comment by NoooneyDude** [comment]
Clients were throwing heaps of errors in the lobby, but editor playmode was fine. Probably just a local issue, could someone else check?

**Comment by ThatDan123** [comment]
Built tested with headless, tested with normal username and trying to join with null and it all seems fine

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**